tariff/entsoe/api.go

```diff
@@ -0,0 +1,140 @@
+// Package entsoe implements a minimalized version of the European Network of Transmission System Operators for Electricity's
+// Transparency Platform API (https://transparency.entsoe.eu)
+package entsoe
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"sort"
+	"time"
+
+	"github.com/dylanmei/iso8601"
+	"github.com/evcc-io/evcc/util/request"
+)
+
+// BaseURI is the root path that the API is accessed from.
+const BaseURI = "https://web-api.tp.entsoe.eu/api"
+
+// numericDateFormat is a time.Parse compliant formatting string for the numeric date format used by entsoe get requests.
+const numericDateFormat = "200601021504"
+
+var ErrInvalidData = errors.New("invalid data received")
+
+// DayAheadPricesRequest represents a helper struct for requesting 4.2.10 Day Ahead Prices [12.1.D]
+type DayAheadPricesRequest struct {
+	domain      string
+	periodStart time.Time
+	periodEnd   time.Time
+}
+
+// ConstructDayAheadPricesRequest constructs a new DayAheadPricesRequest.
+// Domain and duration validity is not checked, that's on you.
+func ConstructDayAheadPricesRequest(domain string, duration time.Duration) DayAheadPricesRequest {
```

Comment:
Wdyt: could we change this to just return the URI? That would eliminate the need to depend on request here and focus only on data representation? Would make this reusable :)

Reply:
Possibly? Usually when building interface packages like these I try to make them return data structs whenever possible, and handle requests inside the package (in this case by passing in the helper). I do see your thinking though.